packages/firestore/rollup.config.js

```diff
@@ -35,16 +35,22 @@ const util = require('./rollup.shared');
 // Customize how import.meta.url is polyfilled in cjs nodejs build. We use it to
 // be able to use require() in esm. It only generates the nodejs version of the
 // polyfill, as opposed to the default polyfill which supports both browser and
-// nodejs. The browser support is unnecessary and doesn't work well with Jest.
+// nodejs. The browser support doesn't work well with Jest.
 // See https://github.com/firebase/firebase-js-sdk/issues/5687
+// Although this is a cjs Node build and shouldn't require the browser option,
+// Vercel apps using this break on deployment, but work in local development.
+// See https://github.com/firebase/firebase-js-sdk/issues/5823
 function importMetaUrlPolyfillPlugin(filename) {
   return {
     name: 'import-meta-url-current-module',
     resolveImportMeta(property, { moduleId }) {
       if (property === 'url') {
-        // copied from rollup output
+        // Copied from rollup output
+        // Added a check for Jest (see issue 5687 linked above)
         return (
-          "(typeof document === 'undefined' ? new (require('url').URL)" +
+          "((typeof document === 'undefined' || process.env.JEST_WORKER_ID " +
```

Comment:
Do we need to check whether `process` and `process.env` are defined?

Reply:
Yes, that seems important, added.